types/react-i18next/src/withNamespaces.d.ts

```diff
@@ -0,0 +1,59 @@
+import * as React from "react";
+import { i18n as I18n, TranslationFunction } from "i18next";
+import { InjectedTranslateProps, InjectedI18nProps } from "./props";
+import { setDefaults, setI18n } from "./context";
+
+export interface TranslateOptions<TTranslateFuncName extends string = string> {
+    bindI18n?: string;
+    bindStore?: string;
+    translateFuncName?: TTranslateFuncName;
+    wait?: boolean;
+    nsMode?: string;
+    i18n?: I18n;
+}
+
+export interface TranslateHocProps {
+    i18n?: I18n;
+    initialI18nStore?: object;
+    initialLanguage?: string;
+}
+
+// Diff / Omit taken from https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-311923766
+type Omit<T, K extends keyof T> = Pick<T, ({ [P in keyof T]: P } & { [P in K]: never } & { [x: string]: never, [x: number]: never })[keyof T]>;
+
+type InjectedProps = InjectedI18nProps & InjectedTranslateProps;
+
+type ComponentPropsWithInnerRef<T extends React.ComponentType<any>> =
+    (T extends React.ComponentClass<infer P>
+        ? P & { innerRef?: React.Ref<InstanceType<T>> }
```

Comment:
See https://github.com/i18next/react-i18next/issues/534

This only works with `withNamespaces` for now.